File: Api/Timesheet/Entities/Vacation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Timesheet.Entities
{
    public class Vacation: EntityBase
    {
        public DateTime InitialDate { get; set; }
        public DateTime EndDate { get; set; }
        public int VacationType { get; set; }
        public string Justification { get; set; }
        public int NumberOfDays { get; set; }
        public DateTime RequestDate { get; set; }

        public  int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
    }
}

File: Web/src/app/models/records/vacation.ts
import { Employee } from './employee';

export class Vacation {
  Id: number;
  EmployeeId: number;
  InitialDate: Date;
  EndDate: Date;
  VacationType: number;
  Justification: string;
  NumberOfDays: number;
  RequestDate: Date;
  Employee?: Employee;

  constructor(params?: Partial<Vacation>) {
    this.Id = params?.Id ;
    this.EmployeeId = params?.EmployeeId ;
    this.InitialDate = params?.InitialDate ;
    this.EndDate = params?.EndDate ;
    this.VacationType = params?.VacationType ;
    this.Justification = params?.Justification ;
    this.NumberOfDays = params?.NumberOfDays ;
    this.RequestDate = params?.RequestDate ;
    this.Employee = params?.Employee ;

  }
}


export enum VacationTypeEnum {
  anuales  = 1,
  maternidad,
  enfermedad ,
  estudios ,
  familiares ,
  duelo  ,
  paternidad  
}